src/test/java/ognl/TestOgnlRuntime.java

```diff
@@ -172,6 +178,111 @@ public void test_Call_Method_VarArgs()
         assertEquals("Halo 3", OgnlRuntime.callMethod(context, service, "getFullMessageFor", args));
     }
 
+    public void test_Call_Method_In_JDK_Sandbox()
+            throws Exception {
+        OgnlContext context = (OgnlContext) Ognl.createDefaultContext(null);
+        GenericService service = new GenericServiceImpl();
+
+        Object[] args = OgnlRuntime.getObjectArrayPool().create(1);
+        args[0] = 0;
+
+        MethodBodyExecutionSandbox.enable(null, null, null);
+
+        try {
+            OgnlRuntime.callMethod(context, service, "exec", args);
+            fail("JDK sandbox should block execution");
+        } catch (Exception ex) {
+            assertTrue(ex.getCause() instanceof SecurityException);
+        } finally {
+            MethodBodyExecutionSandbox.disable();
+        }
+    }
+
+    public void test_Call_Method_In_JDK_Sandbox_Thread_Safety()
+            throws Exception {
+        final OgnlContext context = (OgnlContext) Ognl.createDefaultContext(null);
+        final GenericService service = new GenericServiceImpl();
+
+        MethodBodyExecutionSandbox.enable(null, null, null);
+
+        try {
+            final int NUM_THREADS = 1000;
```

Comment:
**Note of interest**:  It appears this PR's OGNL sandbox may reveal some kind of thread limitation that shows up under JDK 7 (1.7.0_79 32-bit),  The 32-bit JVM threw failures in the test due to inability to create native threads, but it did not happen when not using the sandbox.  The test could complete after dropping the NUM_THREADS below 350 (approximately) under the 32-bit JVM with the sandbox enabled.
Switching to a 64-bit JVM (1.7.0_79 64-bit) made the native thread failures disappear when running under the sandbox.  It could be simply a factor of processing speed or differences in the 32/64-bit JVM implementations ...

Reply:
Nice catch ... I wonder how many users are still based on 32-bit JVM, there is rather a limited number though.

Reply:
:+1: fixed via scaling down by 10.